src/env.rs

```diff
@@ -243,6 +244,21 @@ impl Source for Environment {
             .as_ref()
             .map(|prefix| format!("{prefix}{prefix_separator}").to_lowercase());
 
+        let filter_utf8 = |(key, value): (OsString, OsString)| {
+            let key = if let Ok(key) = key.into_string() {
+                key
+            } else {
+                return None;
+            };
+            let value = if let Ok(value) = value.into_string() {
+                value
+            } else {
+                return None;
+            };
```

Comment:
Does the following work or does it run into type inference issues?
```suggestion
            let key = key.into_string().ok()?;
            let value = value.into_string().ok()?;
```

Reply:
Yes, type inference works, but I think that the changing following your suggestion make it obsolete.